### ISSUE TITLE
Actions don't work in nested component inside `if` or `for` if displayed after first load

### ISSUE THREAD

**Issue by gschurck** [adamghill/django-unicorn#530]
I made a repo to show the issue :

https://github.com/gschurck/django-unicorn-demo

Clicking on Display shows the "Components in if" part, and `unicorn:click` actions is not working in the nested component. And it also does not work in the updated for loop.
A text is printed in console when action is executed.

![image](https://user-images.githubusercontent.com/21091232/227737032-3ecd127e-9ce4-47d4-bf13-e54e313cd723.png)

So it seems like when a component variable is updated and an `if` condition becomes True, no trigger is added on actions buttons for nested components inside this `if`. But it works for actions buttons that are in the `if` but still in the base component. Same case for `for` loop, if the list is updated the actions in new nested component are not working.

**Comment by gschurck** [comment]
@adamghill Do you have any recommendations about this ?

**Comment by adamghill** [comment]
Sorry, I haven't had time to investigate that much, yet. But, thanks for the repo that replicates this issue! I'll try to dig into this as soon as I get some free time to focus on it.

**Comment by gschurck** [comment]
No problem take your time !

I started to investigate. So according to my understanding, [action listener](https://github.com/adamghill/django-unicorn/blob/a98539b6e4b1123705559116a77e63eea7e2b8d0/django_unicorn/static/unicorn/js/eventListeners.js#LL97C30-L97C30) is not working for component displayed after the first page load.

With my demo test, child of the base component are not detected if they are displayed only after a trigger. So when child components are displayed, no action listener is added, and click on buttons does nothing at all.
Components are called from the store `components`, a component is stored during [componentInit](https://github.com/adamghill/django-unicorn/blob/a98539b6e4b1123705559116a77e63eea7e2b8d0/django_unicorn/static/unicorn/js/unicorn.js#L36), and parameter data comes from a [document.getById](https://github.com/adamghill/django-unicorn/blob/a98539b6e4b1123705559116a77e63eea7e2b8d0/django_unicorn/components/unicorn_template_response.py#L144).
A JSON is generated and then added in a script file in the HTML body.

The issue I identified is that this JSON tag was not added for all the children, there is only the base component : 
![image](https://github.com/adamghill/django-unicorn/assets/21091232/381b1be1-f82e-4e0c-89b5-37ca74ad86d9)

And if I change the configuration to display the nested components at the first page load, all the JSON tags of these children are added in the DOM :
![image](https://github.com/adamghill/django-unicorn/assets/21091232/f8ef7a4a-9960-4c5c-a7f5-784fa0594dfc)

So I suppose that if we find a way to update these JSON script tags with the currently displayed components, the rest of the chain should work and action listeners would be added to them ?

**Comment by hendi** [comment]
I can confirm that we also see this issue.

**Comment by imankulov** [comment]
I came here to suggest an alternative, and, hopefully, more complete solution that is outlined in #600 (still an experiment).

I built my solution on top of #538, with the following changes.

- Server: components report their state in a `unicorn:data` atttribute.
- Client: analyze the list of Unicorn components before and after the morph and update the Unicorn.components storage accordingly. Also, we use `querySelectorAll()` which is more performant than manually traversing the node.
